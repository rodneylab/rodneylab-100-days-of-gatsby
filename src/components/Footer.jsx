import React from 'react';

import { ExternalLink } from './Link';

const Footer = () => (
  <footer>
    Page created by
    {' '}
    <ExternalLink
      ariaLabel="Open Rodney Lab contact page"
      to="https://rodneylab.com/contact"
      text="Rodney Lab"
      targetBlank={false}
    />
    {' '}
    as part of
    {' '}
    <ExternalLink
      ariaLabel="See Rodney Lab post on the challenge"
      to="https://rodneylab.com/100-days-of-gatsby-code-2021/"
      text="100 Days of Gatsby Code 2021"
      targetBlank={false}
    />
    .
  </footer>
);

export { Footer as default };
