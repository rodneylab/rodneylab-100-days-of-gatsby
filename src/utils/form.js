import { createClient } from '@formium/client';

export const formium = createClient(process.env.GATSBY_FORMIUM_PROJECTID);

export const validate = (values) => {
  const errors = {};
  if (!values.name) {
    errors.name = 'Please enter your name.';
  }
  if (!values.email) {
    errors.email = 'Please enter your email address.';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Please check your email address.';
  }
  if (!values.message) {
    errors.message = 'Please enter a message.';
  } else if (values.message.length > 1024) {
    errors.message = 'Please limit your message to 1000 characters.';
  }
  return errors;
};
