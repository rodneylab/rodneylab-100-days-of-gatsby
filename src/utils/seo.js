import { getSrc } from 'gatsby-plugin-image';

export const getFeaturedImage = ({ featuredImage }) => ({
  url: getSrc(featuredImage.localFile),
  width: 992,
  height: 730,
});

export const getOgImage = ({ ogImage }) => ({
  url: ogImage ? getSrc(ogImage.localFile) : null,
  width: 1200,
  height: 627,
});

export const getOgSquareImage = ({ ogSquareImage }) => ({
  url: ogSquareImage ? getSrc(ogSquareImage.localFile) : null,
  width: 400,
  height: 400,
});

export const getTwitterImage = ({ twitterImage }) => ({
  url: twitterImage ? getSrc(twitterImage.localFile) : null,
  width: 800,
  height: 418,
});
