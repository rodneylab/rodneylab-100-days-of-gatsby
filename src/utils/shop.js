const formatPrice = (currency, value) => Intl.NumberFormat(undefined, {
  currency,
  minimumFractionDigits: 2,
  style: 'currency',
}).format(value);

export { formatPrice as default };

export const localStorageKey = 'a856f4';
