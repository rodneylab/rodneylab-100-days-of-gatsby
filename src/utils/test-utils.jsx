import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { createMemorySource, createHistory, LocationProvider } from '@reach/router';

const source = createMemorySource('active');
const history = createHistory(source);

const locationProvider = ({ children }) => (
  <LocationProvider history={history}>{children}</LocationProvider>
);

export const customRender = (ui, options) => render(ui, { wrapper: locationProvider, ...options });

export const getMetaByName = (metaName) => {
  const metas = document.getElementsByTagName('meta');
  for (let i = 0; i < metas.length; i += 1) {
    if (metas[i].getAttribute('name') === metaName) {
      return metas[i].getAttribute('content');
    }
  }
  return '';
};

export const getMetaByProperty = (metaProperty) => {
  const metas = document.getElementsByTagName('meta');
  for (let i = 0; i < metas.length; i += 1) {
    if (metas[i].getAttribute('property') === metaProperty) {
      return metas[i].getAttribute('content');
    }
  }
  return '';
};
