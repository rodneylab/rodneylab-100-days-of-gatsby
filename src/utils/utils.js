// eslint-disable-next-line import/prefer-default-export
export const isBrowser = typeof window !== 'undefined';
