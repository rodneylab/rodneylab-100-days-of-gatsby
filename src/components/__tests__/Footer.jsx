import React from 'react';

import Footer from '../Footer';
import { customRender } from '../../utils/test-utils';

describe('ConfusedFaceEmoji', () => {
  it('renders correctly', async () => {
    const elementToRender = (<Footer />);
    const { asFragment } = customRender(elementToRender);
    expect(asFragment(elementToRender)).toMatchSnapshot();
  });
});
