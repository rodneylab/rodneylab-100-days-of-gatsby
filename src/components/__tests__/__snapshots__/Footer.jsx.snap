// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfusedFaceEmoji renders correctly 1`] = `
<DocumentFragment>
  <footer>
    Page created by 
    <a
      aria-label="Open Rodney Lab contact page"
      class="chakra-link css-0"
      href="https://rodneylab.com/contact"
      rel="noopener"
    >
      Rodney Lab 
      <svg
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="2"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <path
            d="M15 3h6v6"
          />
          <path
            d="M10 14L21 3"
          />
        </g>
      </svg>
    </a>
     as part of 
    <a
      aria-label="Open Rodney Lab contact page"
      class="chakra-link css-0"
      href="https://rodneylab.com/100-days-of-gatsby-code-2021/"
      rel="noopener"
    >
      100 Days of Gatsby Code 2021 
      <svg
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="2"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <path
            d="M15 3h6v6"
          />
          <path
            d="M10 14L21 3"
          />
        </g>
      </svg>
    </a>
    .
  </footer>
</DocumentFragment>
`;
